quickwit/quickwit-search/src/root.rs

```diff
@@ -491,6 +491,7 @@ async fn search_partial_hits_phase_with_scroll(
             )
             .next_page(leaf_search_resp.partial_hits.len() as u64);
 
+        scroll_ctx.truncate_start();
```

Comment:
My understanding is that you are calling truncate start to handle the case where the request max hits is > SCROLL_BATCH_LEN. If so, I don't understand why this isn't `truncate` (as opposed to truncate_start) we want to call here.

If this is the use case you are targetting I suggest we :
- do not cache anything if max_hits is too large
- return an error and log a warn if scroll is used with max_hits too large.

Reply:
The idea is that the firsts elements in cache have already been consumed. To not store too many elements (so, to reduce memory pressure), we throw away elements so as to keep at most `SCROLL_BATCH_LEN` elems.
This is arguably not useful, we could keep only one element (to know the values to provide for search_after), and throw everything else

Reply:
Got it.


I'd rather remove the truncate_before. If I understand correctly, it is there to optimize memory.
The complexity added is non-trivial.

The truncation relies on a very bad spec on elasticsearch: scrolling is not nilpotent.
Calling Elasticsearch twice with the same scroll id increases the scroll. (This is bad engineering because it prevents retries on all kinds of errors.)
In addition, the function has a strange behavior that is not suggested in its name. It truncates, but attempts to leave at least one element.

> we throw away elements so as to keep at most SCROLL_BATCH_LEN elems.
Even without the truncation, I think we already have at most `SCROLL_BATCH_LEN` today. (maybe it is a change you made recently).

Reply:
So do we want to keep scrolling idempotent? Before this PR, it was (though at the cost of having each page being slower and more memory intensive to create than the previous).
After, but with the original `truncate_start`, it is idempotent in some cases, but not always (and never if a page is more than `SCROLL_BATCH_LEN`)
With the last commit, it rarely is idempotent